src/_pytest/python.py

```diff
@@ -219,7 +219,12 @@ def pytest_pycollect_makeitem(collector, name, obj):
             )
         elif getattr(obj, "__test__", True):
             if is_generator(obj):
-                res = Generator(name, parent=collector)
+                res = Function(name, parent=collector)
+                res.add_marker(
```

Comment:
I believe this should be a warning, as we discussed previously? I think it is fitting because we also issue warnings for test classes with `__init__`.

Reply:
good call, thanks